### PR TITLE
feat(onyx-1278): restore prompt for app store review for new home view

### PR DESCRIPTION
**Pull request by nickskalkin**
This PR resolves [ONYX-1278]

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#nochangelog

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[ONYX-1278]: https://artsyproduct.atlassian.net/browse/ONYX-1278?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Scenes/HomeView/HomeView.tsx [hunk 1]
```diff
@@ -33,6 +35,8 @@ export const HomeView: React.FC = () => {
 
   const flatlistRef = useBottomTabsScrollToTop("home")
 
+  useMaybePromptForReview({ contextModule: ContextModule.tabBar, contextOwnerType: OwnerType.home })
```
Comment (nickskalkin) [inline src/app/Scenes/HomeView/HomeView.tsx:38]: @dblandin @MounirDhahri I suppose we don't want new context module here?

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ✅ **No changelog changes**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 7ab880d31085b2c456bb7ad7015cb13c80553361
</p>

File: src/app/Scenes/HomeView/HomeView.tsx (hunk 1 above)
Comment (dblandin) [reply src/app/Scenes/HomeView/HomeView.tsx:38]: Right. We'd want to keep the same value here.

**Review by dblandin — DISMISSED** [review]
LGTM 👍🏼 

Not for this PR — but generally I wonder if this sort of trigger belongs up a level and not nested within a particular screen.

The [triggering logic](https://github.com/artsy/eigen/blob/91cbacf316cdac412bb525d3ee90b7ba447b7841/src/app/utils/useMaybePromptForReview.tsx#L8) seems to care about overall app launch count and I don't see anything that is necessarily home screen specific.

cc/ @brainbicycle

**Review by anandaroop — DISMISSED** [review]

File: src/app/Scenes/HomeView/HomeView.tsx (hunk 1 above)
Comment (MounirDhahri) [reply src/app/Scenes/HomeView/HomeView.tsx:38]: yes - that's perfect

**Review by MounirDhahri — APPROVED** [review]

**Comment by brainbicycle** [comment]
> LGTM 👍🏼
> 
> Not for this PR — but generally I wonder if this sort of trigger belongs up a level and not nested within a particular screen.
> 
> The [triggering logic](https://github.com/artsy/eigen/blob/91cbacf316cdac412bb525d3ee90b7ba447b7841/src/app/utils/useMaybePromptForReview.tsx#L8) seems to care about overall app launch count and I don't see anything that is necessarily home screen specific.
> 
> cc/ @brainbicycle

could make sense! only thing we want to make sure of is we only request if logged in!

**Pull request merged**